torch/csrc/autograd/python_function.cpp

```diff
@@ -854,7 +854,15 @@ THPObjectPtr make_ctx_input_output_tuple(
   Py_INCREF(outputs);
   PyTuple_SET_ITEM(result.get(), 0, (PyObject*)ctx);
   PyTuple_SET_ITEM(result.get(), 1, unpacked_input.input_tuple.get());
-  PyTuple_SET_ITEM(result.get(), 2, outputs);
+
+  if (!PyObject_IsInstance(outputs, (PyObject*)(&PyTuple_Type))) {
+    THPObjectPtr wrapped_outputs(PyTuple_New(1));
+    PyTuple_SET_ITEM(wrapped_outputs.get(), 0, outputs);
+    PyTuple_SET_ITEM(result.get(), 2, wrapped_outputs.release());
+  } else {
+    PyTuple_SET_ITEM(result.get(), 2, outputs);
+  }
```

Comment:
Make setup_context consistently return tuple for outputs even if outputs is a single tensor

Reply:
My intended signature was `setup_context(ctx, inputs: Tuple[Any], outputs: Any)`.

The reason is that if we promote outputs to Tuple[Any], then it is confusing to differentiate between the following three cases:
```
def f(x):
  return x

def g(x):
  return (x,)

def h(x):
  return (x, x)
```
If we always promoted outputs to Tuple, then f and g would end up looking the same -- outputs would be (x,) in f and (x,) in g.

Reply:
Hmm that makes sense, I guess my only concern then is calling it outputs plural, which makes it sound like it is a tuple.

Reply:
Change reverted.